tests/Driver/OCI8/ConnectionTest.php

```diff
@@ -0,0 +1,98 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\DBAL\Tests\Driver\OCI8;
+
+use Doctrine\DBAL\Exception\DriverException;
+use Doctrine\DBAL\Tests\FunctionalTestCase;
+use Doctrine\DBAL\Tests\TestUtil;
+use PHPUnit\Framework\Attributes\RequiresPhpExtension;
+use RuntimeException;
+use Throwable;
+
+use function is_resource;
+use function oci_close;
+use function oci_connect;
+use function oci_execute;
+use function oci_parse;
+
+#[RequiresPhpExtension('oci8')]
+class ConnectionTest extends FunctionalTestCase
+{
+    public function tearDown(): void
+    {
+        $this->connection->close();
+    }
+
+    public function testPrepareThrowsErrorOnConnectionLost(): void
+    {
+        $this->expectException(DriverException::class);
+
+        $this->killCurrentSession();
+
+        $this->connection->prepare('SELECT * FROM 1');
+    }
+
+    /**
+     * Get the session details
+     *
+     * @return array<string, mixed>|false
+     */
+    private function getCurrentSession(): array|false
+    {
+        $stmt = $this->connection->prepare("
+            SELECT SID, SERIAL#
+            FROM V\$SESSION
+            WHERE AUDSID = USERENV('SESSIONID')
+        ");
+
+        return $stmt->executeQuery()->fetchAssociative();
+    }
+
+    /**
+     * Kill the current session, by using another connection
+     * Oracle doesn't allow you to terminate the current session, so we use a second connection
+     */
+    private function killCurrentSession(): void
+    {
+        $secondConnection = false;
+
+        try {
+            $session = $this->getCurrentSession();
+            if ($session === false || ! isset($session['SID'], $session['SERIAL#'])) {
+                return;
+            }
+
+            $params   = TestUtil::getConnectionParams();
+            $username = $params['user'] ?? '';
+            $password = $params['password'] ?? '';
+            $host     = $params['host'] ?? '';
+
+            $secondConnection = oci_connect($username, $password, $host);
+            if ($secondConnection === false) {
```

Comment:
Please use DBAL, if possible. I feel like the amount of boilerplate code will reduce.

Reply:
Done